### ISSUE TITLE
Permission errors with tools container image on ubuntu host

### ISSUE THREAD

**Issue by BlueHatbRit** [khuedoan/homelab#37]
Hi, huge fan of this project! I've been studying it for a few evenings to understand how I might adapt this for my home lab which is currently 2 Lenovo ThinkCentre M700's. My workstation OS is Ubuntu and I've been having some issues with the tools container, it seems while it'll boot into the container as the same user, etc, the container then has no permissions to a bunch of files around my system.

Some of the issues I've run into so far include:
* Issues accessing ~/.ssh/config and keys
* "Destination is not writable" when ansible does the get_url on the rocky linux ISO to the `metal/roles/pxe_server/files/data/iso` directory.

I've managed to get through some of these so far by converting the Dockerfile over to a Ubuntu base image and getting all the dependencies setup but the ansible issue above is still causing problems. I'm just curious, are you running on Arch like the container images is? I'm wondering if there are some differences between Ubuntu and Arch which might be causing the problem and will give me somewhere to look.

If so, it might also be worth flagging this in the docs / readme that "YMMV if you're not on arch". I'm going to continue trying to get things working, but wanted to check my assumption of your workstation OS first. If that is the problem, I might look at trying it in an Arch VM to dig a bit deeper.

**Comment by khuedoan** [comment]
The host OS shouldn't matter, I've tested it on Arch and Rocky, some people use Debian and Ubuntu as well. The tools container will mount some volumes and run as the same UID/GID as the user on the host.

What is the current permissions of the folders you're having issue with?

**Comment by BlueHatbRit** [comment]
Oooh okay sounds like it's just me having the issue then. On the arch Dockerfile image it's unable to access `~/.ssh` and any config and key files. Everything in there is chmod 600. I was unable to get past that without opening up the permissions to 777, so at that point I decided to make some changes to the Dockerfile. I did notice the Make command which is what confused me and why I thought it was potentially some quirk of docker not liking the different distro / base image.

As soon as I used the same docker run command with a ubuntu base image it was happy on the ssh path which was what lead me to wonder if this only worked for those with an arch host.

**Comment by khuedoan** [comment]
Maybe something changed in those folder in the mean time, does it work with the included container now?

**Comment by BlueHatbRit** [comment]
Hey, sorry for the delay, I wanted some proper time to go through this and make sure I wasn't going crazy. Turns out I'm not going crazy, but I am flat out stupid :laughing:

The uid, and gid are both correct and updating to the latest pull didn't change anything, but then I realise I was running podman and not docker :facepalm: it had been working so seamlessly for so long I completely forgot.

I uninstalled podman, walked through everything again and it's working flawlessly with no problems. It would seem podman is doing something subtly different which is  causing the file system permission here.

I'll go ahead and close this one now! Thanks for your responsiveness and time thinking about this. Sorry for wasting your time with this one. I guess the consolation is we know it needs docker and not podman :sweat:
